universal-application-tool-0.0.1/test/controllers/applicant/ApplicantProgramBlocksControllerTest.java

```diff
@@ -232,4 +284,22 @@ public void edit_withMessages_returnsCorrectButtonText() {
     assertThat(result.status()).isEqualTo(OK);
     assertThat(contentAsString(result)).contains("Guardar y continuar");
   }
+
+  private Applicant createApplicantWithMockedProfile() {
+    Applicant applicant = resourceCreator.insertApplicant();
+    Account account = resourceCreator.insertAccount();
+
+    account.setApplicants(ImmutableList.of(applicant));
+    account.save();
+    applicant.setAccount(account);
+    applicant.save();
+
+    UatProfile profile = profileFactory.wrap(applicant);
+    mockProfile(profile);
+    return applicant;
+  }
+
+  private void mockProfile(UatProfile profile) {
+    when(MOCK_UTILS.currentUserProfile(any(Http.Request.class))).thenReturn(Optional.of(profile));
+  }
```

Comment:
I think you're right, a browser test probably makes more sense. We have to mock or do something for these controller tests anyway, since leaving them as-is means they all fail (none have profiles set by default). 

I'll add a browser test and update

Reply:
Ok I added a browser test! It just tests if we can reach an applicant endpoint as an admin vs. applicant  - mind taking another look?

Reply:
Ah, forgot this would fail without the change. Thanks for adding the browser tests!